server/src/test/java/org/elasticsearch/env/NodeMetadataTests.java

```diff
@@ -106,16 +112,32 @@ public void testDoesNotUpgradeAncientVersion() {
         );
         assertThat(
             illegalStateException.getMessage(),
-            allOf(startsWith("cannot upgrade a node from version ["), endsWith("] directly to version [" + Version.CURRENT + "]"))
+            allOf(
+                startsWith("cannot upgrade a node from version ["),
+                endsWith(
+                    "] directly to version ["
+                        + Version.CURRENT
+                        + "], upgrade to version ["
+                        + Version.CURRENT.minimumCompatibilityVersion()
+                        + "] first."
+                )
+            )
         );
     }
 
-    public void testUpgradeMarksPreviousVersion() {
```

Comment:
I think this is the only test that asserts that `previousNodeVersion` works, so we should keep it around.